cc_configure_make/def.bzl

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _cc_configure_make_impl(ctx):
+    out_includes = ctx.actions.declare_directory(ctx.attr.name + "-includes.h")
+    out_lib = ctx.actions.declare_file("{}.a".format(ctx.attr.name))
+    outputs = [out_includes, out_lib]
+
+    cpp_fragment = ctx.fragments.cpp
+    compiler_options = [] # cpp_fragment.compiler_options(ctx.features)
+    c_options = compiler_options + cpp_fragment.c_options
+    cxx_options = compiler_options + cpp_fragment.cxx_options(ctx.features)
+
+    CFLAGS = "\"{}\"".format(' '.join(c_options))
+    CXXFLAGS = "\"{}\"".format(' '.join(cxx_options))
```

Comment:
FYI, but there's a *lot* of env vars that we should support (https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html). We will add them as we go.

Reply:
Hmm but we should specify `CC` and `CXX` at least, and once https://docs.google.com/document/u/1/d/1g91BWJITcYw_X-VxsDC0VgUn5E9g0kRBGoBSpoO41gA/edit is done (O(weeks)) we can specify different tools for those

Reply:
How will the code look like here?

Reply:
roughly like this: cc_common.configure_features(cc_toolchain).get_command_line(cc_common.c_compile_action_name, cc_common.get_compile_variables(....))